flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/calcite/SqlRewriterUtils.scala

```diff
@@ -193,11 +193,33 @@ object SqlRewriterUtils {
     // Expands the select list first in case there is a star(*).
     // Validates the select first to register the where scope.
     validator.validate(cte)
-    val select = cte.body.asInstanceOf[SqlSelect];
-    reorderAndValidateForSelect(validator, select, targetRowType, assignedFields, targetPosition)
+    val selects = new util.ArrayList[SqlSelect]()
+    extractSelectsFromCte(cte.body.asInstanceOf[SqlCall], selects)
+
+    for (select <- selects) {
+      reorderAndValidateForSelect(validator, select, targetRowType, assignedFields, targetPosition)
+    }
     cte
   }
 
+  def extractSelectsFromCte(cte: SqlCall, selects: util.List[SqlSelect]): Unit = {
```

Comment:
Nice!  

I kinda don't like that `selects` is passed in instead of returned.  I tried refactoring it, but it doesn't look much better.